sdk/storage/storage-blob/test/retrypolicy.spec.ts

```diff
@@ -123,63 +125,60 @@ describe("RetryPolicy", () => {
 
   it("Retry Policy should work for secondary endpoint", async () => {
     let injectCounter = 0;
-    const injector = new InjectorPolicyFactory(() => {
+    const injector = injectorPolicy(() => {
       if (injectCounter++ < 1) {
-        return new RestError("Server Internal Error", "ServerInternalError", 500);
+        return new RestError("Server Internal Error", {
+          code: "ServerInternalError",
+          statusCode: 500,
+        });
       }
       return;
     });
 
     const url = blobServiceClient.url;
-    const urlParsed = URLBuilder.parse(url);
-    const host = urlParsed.getHost()!;
+    const urlParsed = new URL(url);
+    const host = urlParsed.hostname;
     const hostParts = host.split(".");
     const account = hostParts.shift();
     const secondaryAccount = `${account}-secondary`;
     hostParts.unshift(secondaryAccount);
     const secondaryHost = hostParts.join(".");
 
-    const credential = (containerClient as any).pipeline.factories[
-      (containerClient as any).pipeline.factories.length - 1
-    ];
-    const factories = newPipeline(credential, {
-      retryOptions: { maxTries: 2, secondaryHost },
-    }).factories;
-    factories.push(injector);
-    const pipeline = new Pipeline(factories);
-    const injectContainerClient = new ContainerClient(containerClient.url, pipeline);
+    blobServiceClient = getBSU({ retryOptions: { maxTries: 2, secondaryHost } });
+    containerClient = blobServiceClient.getContainerClient(containerName);
+
+    const pipeline: Pipeline = (containerClient as any).storageClientContext.pipeline;
+    pipeline.addPolicy(injector, { afterPhase: "Retry" });
 
     let finalRequestURL = "";
     try {
-      const response = await injectContainerClient.getProperties();
+      const response = await containerClient.getProperties();
       finalRequestURL = response._response.request.url;
     } catch (err: any) {
       finalRequestURL = err.request ? err.request.url : "";
     }
 
-    assert.deepStrictEqual(URLBuilder.parse(finalRequestURL).getHost(), secondaryHost);
+    assert.deepStrictEqual(new URL(finalRequestURL).hostname, secondaryHost);
   });
 
   it("Retry Policy should work when on PARSE_ERROR with unclosed root tag", async () => {
     let injectCounter = 0;
-    const injector = new InjectorPolicyFactory(() => {
+    const injector = injectorPolicy(() => {
```

Comment:
We may keep this case to test its compatibility.

Reply:
I'm not sure I follow, but are you suggesting we add a test case with an old-style PolicyFactory to ensure compatibility of user defined policies? That seems like a good idea since today we get some coverage from the SDKs that depend on storage-blob, but after those are migrated this will go down.